doc/srv6/srv6_policy.md

```diff
@@ -0,0 +1,511 @@
+
+```
+
+Zebra collapses these two levels into one level which would be described below. With NHG ID support, we will have NHG contains a set of NH IDs, which point to candidate paths' nexthop IDs, a.k.a SRv6 Tunnel Nexthop  IDs. This part is the same as normal NHG ID handling. One new field "weight" would be added to each path of NHG. This weight is used for forming WCMP. 
+
+```
+; Modify table
+; holds NEXTHOP_GROUP_TABLE
+
+key = NEXTHOP_GROUP_TABLE|<nhgid>
+
+; field = value
+weight =  <weight>,   ; 
+```
+
+A new field, "segment",  would be added in nexthop schema. This field is used in candidate path's nexthop for providing sid list name. 
+
+```
+; Modify table
+; holds NEXTHOP_TABLE
+
+key = NEXTHOP_TABLE|<nhgid>
+
+; field = value
+segment =  <seg name>,   ;  sid list name
+
+
+```
+
+## SID LIST Schema
+SID LIST would be populated from pathd. This schema provides detail path information for each candidate path, a.k.a SID LIST.
+
+Here is an example for SID_LIST in appdb. 
+```
+; New table
+; holds SRV6_SID_LIST_TABLE
+
+key = SRV6_SID_LIST_TABLE|<sid_list_name>
+
+; field = value
+path= ip addresses,   ; List of segment addresses, seperated via ,
+
+For example:
+
+127.0.0.1:6380> keys *SID_LIST*
+1) "SRV6_SID_LIST_TABLE:slist_b"
+2) "SRV6_SID_LIST_TABLE:slist_d"
+3) "SRV6_SID_LIST_TABLE:slist_a"
+4) "SRV6_SID_LIST_TABLE:slist_c"
+127.0.0.1:6380> hgetall "SRV6_SID_LIST_TABLE:slist_a"
+1) "path"
+2) "fd00:205:205:fff5:55::"
+127.0.0.1:6380> 
+```
+
+# Color Only Policy use case
+The policy is determined by color only. Implicitly it means the endpoint address is ::/0. This kind of policy's UP/DOWN state is determined by candidate paths' BFD. Routes would use color attribute only to resolve to this policy. This kind of policy is servered as a default route for the given color. 
+
+# Candidate-path management in pathd
+Similar to MPLS SR cases, Pathd is responsible to maintain each candidate path's status. BFD protection on candidate paths would be discussed in a separate HLD.  From zebra point of view, it would get ZEBRA_SRV6_POLICY_SET message from pathd whenever a new policy is applied or some policy information is updated. Zebra would get ZEBRA_SRV6_POLICY_DELETE message from pathd whenever a policy is deleted. Policy message carries all used SID LISTs. 
+
+## Candidate path's state machine
+## Candidate path without bfd configuration
+In this case, the candidate path's default state is UP. Pathd would not be able to manage cpath's state. It would be external controller's job to manage cpath's state.
+
+## Candidate path with bfd configuration
+In this case, the default state is DOWN. bfd notification would be used to update cpaths' state. 
+
+### Policy with endpoint
+bfd packet would be constructed with SRv6 header which would go via both specified cpath and engpoint. The protection path would be via specified SID notes and the end point node. This protection would not cover link if the SID list is not hop by hop. 
```

Comment:
Does it make use of h/w offloaded BFD?

Reply:
Both software and hw offload, depending on BFD timer and scale.